apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceExtension.cs

```diff
@@ -95,25 +105,35 @@ public static void UseGoogleTrace(this IApplicationBuilder app)
         /// <param name="projectId">The Google Cloud Platform project ID. Cannot be null.</param>
         /// <param name="config">Optional trace configuration, if unset the default will be used.</param>
         /// <param name="client">Optional Trace client, if unset the default will be used.</param>
+        /// <param name="shouldTraceFunc">Optional function to trace requests. If the trace header is not set
+        ///     then this function will be called to determine if a given request should be traced.  This will
+        ///     not override trace headers. If the function returns true the request will be traced, if false
+        ///     is returned the trace will not be traced and if null is returned it will not affect the
+        ///     trace decision.</param>
         public static void AddGoogleTrace(
             this IServiceCollection services, string projectId,
-            TraceConfiguration config = null, TraceServiceClient client = null)
+            TraceConfiguration config = null, TraceServiceClient client = null,
+            Func<HttpRequest, bool?> shouldTraceFunc = null)
```

Comment:
Possibly rename to traceOverridePredicate? Definitely put Predicate in the name to mimic LINQ. (Admittedly the fact that it can return null makes it slightly different, but only slightly.)